### ISSUE TITLE
CoDe-Academy takeaways

### ISSUE THREAD

**Issue by sofusalbertsen** [Praqma/reveals#44]
When are they going to set up AWS/GIT/Putty and all that jazz?
3 hours is too long for LEGO. 1,5 hours is apropriate, and then maybe more teaching about agile task management?

**Comment by sofusalbertsen** [comment]
Name tags for all.
Prerequisites: install git + putty + whatever else (ssh keys in GH).

**Comment by sofusalbertsen** [comment]
Make small slideset for LEGO game, describing the teams, the roles, and the sprints. The game rules.

**Comment by sofusalbertsen** [comment]
Jenkins too short for 1,5 hours if they need to install jenkins and make connection to git as well. Could use more time, or maybe install Jenkins the day before. (make a script for them, because they do not get what they do anyway)

**Comment by sofusalbertsen** [comment]
Forget about Putty, use git bash.

**Comment by sofusalbertsen** [comment]
Need to take more of day 1 to tech things. Otherwise we basically have 2 days to 4 technologies.

**Comment by sofusalbertsen** [comment]
"you should remove the cigarette on the slides"
We need to make the marker in reveals template look less like a cigarette ;)

**Comment by sofusalbertsen** [comment]
We did not get into PiP, so we need to have focus on that.

**Comment by kvarak** [comment]
**_Some observations from CoDe Academy Göteborg._** 

**Supergreat things!**

* Usage of Slack
* Energy of the teachers
* Knowledge of the teachers
* kitkatas on github
* The project

**Regarding the exercises**

The level of knowledge among the students are so different so that the brighter ones always need to wait for the "slower" ones to catch up (or almost catch up). It would be good to have something extra that could be worked on while waiting. Some sat a waited, others started to do other things. Maybe always have a couple of extra gitkatas listed. For instance:

>Do these:
>gitkata/easy
>gitkata/kinda-easy
>gitkata/not-so-easy
>
>If you have time, you can continue with these:
>gitkata/hard
>gitkata/super-hard

This will always keep everyone busy, but make sure that the "easy" ones will have exercises will exercise everything so far presented by the teacher, so that the "slower" ones don't miss out.

**Regarding the test part**

Emily did a really good exercise and presentation, I have no comments on that. However, the ones I talked to wanted to have the presentation first to understand the *why*.

Many of the ones I talked to don't actually learn anything in school, their education is *"problem-centric"*, where they are presented a problem, the tools to use to fix the problem and then they are pretty much on their own (plus google and a lab assistant that doesn't really care) to use the tools to figure out the correct way.

For instance: "Solve this, use git and Java, collaborate with somebody else"

Ok, maybe I have the wrong idea, maybe it's not this bad and I'm making it sound like schools are shit. But the point is that they have **no** idea how to use testing while coding in a proper way. So this section is so new to them that they have a really hard time understanding the *why*.

TL;DR; more test-topics
(sorry, couldn't stop writing after I started).

**Comment by sofusalbertsen** [comment]
Gilded rose for the teaching days, included in test, jenkins, maybe in docker as well.
https://github.com/emilybache/GildedRose-Refactoring-Kata

**Comment by sofusalbertsen** [comment]
@kvarak I've tried to take your feedback into the trello retrospective:
https://trello.com/b/EdNntcrC/code-academy-retrospective-2017
will close this one.
